### PR TITLE
`feat(TCK-TS):` Migrate TCK to TypeScript for Improved Maintainability

### PR DESCRIPTION
**Pull request by ivaylogarnev-limechain**
**Description**:
This PR migrates the TCK project from `JavaScript` to `TypeScript`, enhancing the codebase with type safety, better structure, and improved maintainability. The migration is aimed at making the project more developer-friendly and reducing runtime errors by leveraging TypeScript's static type-checking capabilities.

**Related issue(s)**:
 #275

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by 0xivanov — COMMENTED** [review]
I would keep the test-specifications outside the src folder, what do you think?

File: src/tests/account-service/test-account-create-transaction.ts [hunk 1]
```diff
@@ -244,7 +245,7 @@ describe("AccountCreateTransaction", function () {
           key: key.key,
           initialBalance: -1,
         });
-      } catch (err) {
+      } catch (err: any) {
```
[the remark this answers is not in the document]
Comment (ivaylogarnev-limechain) [reply src/tests/account-service/test-account-create-transaction.ts:248]: `TypeScript` requires that the catch clause variable type annotation should be either `any` or `unknown`.  

I've tried it an if we want to add such an error we should do something like

```
} catch (err: unknown) {
  if (err instanceof HederaError) {
    // Handle HederaError specifically
  } else {
    // Handle other errors
  }
}
```

However, I believe this adds unnecessary complexity to our error handling which in my opinion is not needed for our case.

File: src/tests/account-service/test-account-create-transaction.ts [hunk 2]
```diff
@@ -60,12 +62,12 @@ describe("AccountCreateTransaction", function () {
...
 
-    it("(#2) Creates an account with a valid ECDSAsecp256k1 public key", async function () {
+    it("(#2) Creates an account with a valid ECDSAsecp256k1 public key", async () => {
       // Generate an ECDSAsecp256k1 public key for the account.
       //prettier-ignore
       const ecdsaSecp256k1PublicKey = await JSONRPCRequest(this, "generateKey", {
-        type: "ecdsaSecp256k1PublicKey",
-      });
+      type: "ecdsaSecp256k1PublicKey",
+    });
```
[the remark this answers is not in the document]
Comment (ivaylogarnev-limechain) [reply src/tests/account-service/test-account-create-transaction.ts:70]: There was an issue with Prettier configuration, you might need to configure it locally. Please refer to the `.prettierrc` file located in the current directory, you need to specify within the VSCode folder under `settings.json`:

```
{
  "prettier.configPath": "./.prettierrc"
}
```

**Comment by ivaylogarnev-limechain** [comment]
> Couple things:
> 
> * Why the rename of `crypto-service` to `account-service`? `crypto-service` is how it is referred to in the protobufs.
> * I like the change of the file names from `test_<requestType>.js` to `type-<request-type>.ts, how do you feel about doing the same thing to the test specification markdown files? Also, `DevelopmentProcess.md` describes the naming conventions of the files so that should be updated.

1. Ah, I understand the logic behind it now – reverted.
2. I might prefer to keep the `.md` files in **camelCase**, and to be perfectly correct, we need to capitalize the first letter. The reason for this is to make it easier to differentiate between the `.md` (documentation) and `.ts` (tests) files when searching within the directory. @0xivanov WDYT?

**Comment by 0xivanov** [comment]
@ivaylogarnev-limechain I'm for making the md files camel case with capital letter.

**Comment by rwalworth** [comment]
> > I would keep the test-specifications outside the src folder, what do you think?
> 
> I was also wondering where should I put them, I also think it's a good idea to keep the test specifications outside the src (logic) folder.
> 
> @rwalworth WDYT?

I think just keeping them in the root directory is fine. Or maybe we can throw them in a `docs` folder? Could throw `DevelopmentProcess.md` in there as well. I'm fine either way

**Review by 0xivanov — APPROVED** [review]
LGTM

**Pull request merged**